src/CMakeLists.txt

```diff
@@ -115,6 +115,12 @@ elseif(CMAKE_BUILD_TYPE STREQUAL "Release" OR CMAKE_BUILD_TYPE STREQUAL "RelWith
     )
 endif()
 
+add_compile_definitions(
+	# Whether to validate narrowing conversions done with narrow_cast<T>(...)
+	# If not, will simply alias narrow_cast to static_cast
+	$<$<CONFIG:Debug>:PARANOID_NARROWING>
```

Comment:
It's better to put this define under `if(CMAKE_BUILD_TYPE STREQUAL "Debug")` rather than using generator expression so IDE can know this and highlight the corresponding `#ifdef` branch.